### PR TITLE
fix(argo-workflows): add `app:` label to components to match upstream

### LINKED ISSUE #2774: Workflow controller pod labels do not match upstream
### Describe the bug

The Workflow Controller deployment in the official release manifests get one label on the resulting pod: `app: workflow-controller`
The Workflow Controller deployment in the helm chart gets these labels:
```
app.kubernetes.io/name: {{ include "argo-workflows.name" .context }}-{{ .name }}
app.kubernetes.io/instance: {{ .context.Release.Name }}
app.kubernetes.io/component: {{ .component }}
```

is there a historical reason why the helm chart differs from upstream?

Any objections if I added app: workflow-controller to the helm chart (probably taking `.component` as the value?

### Related helm chart

argo-workflows

### Helm chart version

0.41.10

### To Reproduce

Install.
Stare at the resulting labels on the controller pod.

### Expected behavior

I would expect the helm chart to match upstream where possible.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tico24**
Closes #2774

Here's the resulting labels on the controller pod when running a default install on this pr:
```yaml
  template:
    metadata:
      labels:
        helm.sh/chart: argo-workflows-0.41.10
        app.kubernetes.io/name: argo-workflows-workflow-controller
        app.kubernetes.io/instance: release-name
        app.kubernetes.io/component: workflow-controller
        app: workflow-controller
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/part-of: argo-workflows
        app.kubernetes.io/version: "v3.5.8"
    spec:
```

Nothing has been removed so I don't believe this is a breaking change.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by yu-croco — APPROVED** [review]
Great catch! 🚀

File: renovate.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "**/charts/argo-events/Chart.yaml",
     "**/charts/argo-rollouts/Chart.yaml",
     "**/charts/argocd-image-updater/Chart.yaml",
-    "**/.github/workflows/renovate.yaml"
+    "**/.github/workflows/renovate.yaml",
+    "**/.github/workflows"
```
Comment (mkilchhofer) [inline renovate.json:18]: Does that mean that we now enable renovate on all workflow files?

Comment (tico24) [reply renovate.json:18]: Oh no. That's a bad commit. I'll remove when I'm at a computer.

Comment (tico24) [reply renovate.json]: The badness has been removed now. Sorry about that.

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**